src/components/Search/SearchPageHeaderInput.tsx

```diff
@@ -0,0 +1,273 @@
+    switch (type) {
+        case CONST.SEARCH.DATA_TYPES.INVOICE:
+            return {icon: Illustrations.EnvelopeReceipt, titleText: 'workspace.common.invoices'};
+        case CONST.SEARCH.DATA_TYPES.TRIP:
+            return {icon: Illustrations.Luggage, titleText: 'travel.trips'};
+        case CONST.SEARCH.DATA_TYPES.CHAT:
+            return {icon: Illustrations.CommentBubblesBlue, titleText: 'common.chats'};
+        case CONST.SEARCH.DATA_TYPES.EXPENSE:
+        default:
+            return {icon: Illustrations.MoneyReceipts, titleText: 'common.expenses'};
+    }
+}
+
+function SearchPageHeaderInput({queryJSON, children}: SearchPageHeaderInputProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const personalDetails = usePersonalDetails();
+    const [reports] = useOnyx(ONYXKEYS.COLLECTION.REPORT);
+    const taxRates = useMemo(() => getAllTaxRates(), []);
+
+    const [autocompleteSubstitutions, setAutocompleteSubstitutions] = useState<SubstitutionMap>({});
+    // The actual input text that the user sees
+    const [textInputValue, setTextInputValue] = useState(' '); // initial empty space to avoid quick flash of placeholder text
+    // The input text that was last used for autocomplete; needed for the SearchRouterList when browsing list via arrow keys
+    const [autocompleteQueryValue, setAutocompleteQueryValue] = useState(textInputValue);
+
+    const [isAutocompleteListVisible, setIsAutocompleteListVisible] = useState(false);
+    const listRef = useRef<SelectionListHandle>(null);
+
+    const {type, inputQuery: originalInputQuery} = queryJSON;
+    const isCannedQuery = SearchQueryUtils.isCannedSearchQuery(queryJSON);
+    const headerText = isCannedQuery ? translate(getHeaderContent(type).titleText) : '';
+    const queryText = SearchQueryUtils.buildUserReadableQueryString(queryJSON, personalDetails, reports, taxRates);
+
+    useEffect(() => {
+        setTextInputValue(queryText);
+    }, [queryText]);
+
+    useEffect(() => {
+        const substitutionsMap = buildSubstitutionsMap(originalInputQuery, personalDetails, reports, taxRates);
+        setAutocompleteSubstitutions(substitutionsMap);
+    }, [originalInputQuery, personalDetails, reports, taxRates]);
+
+    const onSearchQueryChange = useCallback(
+        (userQuery: string) => {
+            const updatedUserQuery = SearchAutocompleteUtils.getAutocompleteQueryWithComma(textInputValue, userQuery);
+            setTextInputValue(updatedUserQuery);
+            setAutocompleteQueryValue(updatedUserQuery);
+
+            const updatedSubstitutionsMap = getUpdatedSubstitutionsMap(userQuery, autocompleteSubstitutions);
+            setAutocompleteSubstitutions(updatedSubstitutionsMap);
+
+            if (updatedUserQuery || textInputValue.length > 0) {
+                listRef.current?.updateAndScrollToFocusedIndex(0);
+            } else {
+                listRef.current?.updateAndScrollToFocusedIndex(-1);
```

Comment:
The `updatedUserQuery || textInputValue.length > 0)` is pretty much always true because textInputValue is stale and holds the previous value. This caused the focus not being cleared on clearing the text input.